File: _scratch/ScottStuff/en-us/channel-inspector/files/Skype/Typing/example1.cs
public Task ExecuteCommand(ActivityContext activityContext, Activity activity)
{
	Background.Run(async () =>
	{
		IMessageActivity iMessage = (IMessageActivity)activity;
		var reply = activity.CreateReply(String.Empty);
		reply.Type = ActivityTypes.Typing;
		await SendResponse(activityContext, reply).DropContext();
		await Task.Delay(2000).DropContext();
		await SendResponse(activityContext, reply).DropContext();
		await Task.Delay(3000).DropContext();
		reply.Type = ActivityTypes.Message;
		reply.Text = "My robot must rate as my favorite toy,  A wonderful, whirring, mechanical joy.  My robot can talk, but he'd much rather sing,  Or go to the park and play on the swings!";
		await SendResponse(activityContext, reply).DropContext();
	});
	
	return Task.CompletedTask;
}

File: _scratch/ScottStuff/en-us/channel-inspector/files/Email/Typing/example1.js
function BotSendTyping(session, builder){
	function delaySend(index) {
		setTimeout(function () {
			if(index < 3){
				if(index == 0){
					var msg = new builder.Message(session)
					.text("");
					session.sendTyping();
					session.send(msg);
				}
				else if(index == 1){
					var msg = new builder.Message(session)
					.text("");
					session.sendTyping();
					session.send(msg);
				}
				else if(index == 2){
					var msg = new builder.Message(session)
					.text("My robot must rate as my favorite toy,  A wonderful, whirring, mechanical joy.  My robot can talk, but he'd much rather sing,  Or go to the park and play on the swings!");
					session.send(msg);					
				}
				delaySend(index + 1);
			}
			else{
				//Do something
			}
		}, 2000);		
	}
	delaySend(0);	
}
